src/event-engine/index.ts

```diff
@@ -136,11 +136,11 @@ export class EventEngine {
   }
 
   getSubscribedChannels(): string[] {
-    return this.channels.slice(0);
+    return Array.from(new Set(this.channels.slice(0)));
   }
 
   getSubscribedChannelGroups(): string[] {
-    return this.groups.slice(0);
+    return Array.from(new Set(this.groups.slice(0)));
```

Comment:
Do we really need `slice(0)` if there is only primitive “string”?
```suggestion
    return Array.from(new Set(this.groups));
```
If there were an object or another array - then yup, we will need to make a “copy” of it.